tron/utils/scribereader.py

```diff
@@ -90,146 +141,108 @@ def read_log_stream_for_action_run(
     if min_date is None:
         return [f"{action_run_id} has not started yet."]
 
+    use_s3_reader = False
+
+    if S3LogsReader:
+        config_watcher = get_config_watcher()
+        config_watcher.reload_if_changed()
+        use_s3_reader = staticconf.read("logging.use_s3_reader", namespace=NAMESPACE, default=False)
+    elif scribereader is None:
+        return ["Neither scribereader nor yelp_clog (internal Yelp packages) are available - unable to display logs."]
+
     if max_lines == USE_SRV_CONFIGS:
         config_watcher = get_config_watcher()
         config_watcher.reload_if_changed()
         max_lines = staticconf.read("logging.max_lines_to_display", namespace=NAMESPACE)
 
-    if scribereader is None:
-        return ["Scribereader (an internal Yelp package) is not available - unable to display logs."]
-    if get_scribereader_host_and_port() is None:
+    try:
+        ecosystem = get_ecosystem()
+        superregion = get_superregion()
+        region = get_region()
+    except OSError:
+        log.warning("Unable to read location mapping files from disk, not returning scribereader host/port")
         return [
             "Unable to determine where Tron is located. If you're seeing this inside Yelp, report this to #compute-infra"
         ]
-    host, port = get_scribereader_host_and_port()  # type: ignore  # the None case is covered by the check above
 
-    # this should never fail since get_scribereader_host_and_port() will have also called get_superregion() and we've ensured that
-    # that file exists by getting to this point
     if paasta_cluster is None:
-        paasta_cluster = get_superregion()
+        paasta_cluster = superregion
+
+    paasta_logs = PaaSTALogsIterator(component, paasta_cluster, action_run_id)
```

Comment:
i also wonder if we should rename this class since it's not returning an iterator

Reply:
valid point, it's definitely not an iterator but rather does the "iteration"